src/components/ArticleByTopic.jsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from "react";
+import { getTopics } from "../utils/api";
+import { Link } from "react-router-dom";
+
+const ArticleByTopic = () => {
+  const [error, setError] = useState();
+  const [topics, setTopics] = useState();
+
+  useEffect(() => {
+    getTopics()
+      .then((topics) => {
+        setTopics(topics);
+      })
+      .catch((err) => {
+        setError(err.response.data);
+      });
+  }, []);
+
+  if (!topics) {
+    return <h1>{error}</h1>;
+  }
+
+  return (
+    <>
+      <main>
+        <ul>
+          {topics.map((topic) => {
+            return (
+              <>
+                <li key={topic.slug}>
+                  <Link to={`/topics/${topic.slug}`}>
+                    <h2>{topic.slug.toUpperCase()}</h2>
+                  </Link>
+                  <p>{topic.description}</p>
+                </li>
+              </>
+            );
+          })}
+        </ul>
+      </main>
+    </>
+  );
+};
+
+export default ArticleByTopic;
```

Comment:
I don't think ArticleByTopic really describes what this component is! I'd probably suggest calling it Topics?

Reply:
Done